pkg/syncer/cnsoperator/util/util.go

```diff
@@ -188,3 +191,52 @@ func GetNetworkProvider(ctx context.Context) (string, error) {
 	return "", fmt.Errorf("could not find network provider field in configmap %q in namespace %q",
 		wcpNetworkConfigMap, kubeSystemNamespace)
 }
+
+// GetConfig is a wrapper function in syncer container to get the
+// config from vSphere Config Secret. If cluster ID is not provided
+// in the secret, then we read it from the immutable ConfigMap
+// which was created during csi controller initialization.
+func GetConfig(ctx context.Context) (*cnsconfig.Config, error) {
+	var clusterID string
+	log := logger.GetLogger(ctx)
+
+	cfg, err := common.GetConfig(ctx)
+	if err != nil {
+		log.Errorf("failed to read config. Error: %+v", err)
+		return nil, err
+	}
+
+	if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.CSIInternalGeneratedClusterID) &&
+		cfg.Global.ClusterID == "" {
+		CSINamespace := common.GetCSINamespace()
+
+		cmData, err := commonco.ContainerOrchestratorUtility.GetConfigMap(ctx,
+			cnsconfig.ClusterIDConfigMapName, CSINamespace)
+		if err == nil {
+			// Get the clusterID value stored in the existing immutable ConfigMap.
+			clusterID = cmData["clusterID"]
+			log.Infof("Cluster ID value read from the ConfigMap is %s", clusterID)
+		} else {
+			return nil, logger.LogNewErrorf(log, "cluster ID is not available in "+
+				"vSphere config secret as well as in immutable ConfigMap")
```

Comment:
vSphere config secret and in immutable ConfigMap

Reply:
Done